### DISCUSSION TITLE
Is there a way to auto create migrations from existing tables?

### DISCUSSION THREAD

**Discussion by IDrumsey** [sequelize/sequelize#17277] [Help & Questions]
If you have a table that exists in a database, but wasn't created with a migration, is there any way with sequelize/sequelize-cli to create a migration for that table without having to do it manually?

**Comment by GreenFlag31** [comment]
npx sequelize-cli migration:create --name name-of-the-file to generate a migration file for a given database
npx sequelize-cli model:generate --name User --attributes firstName:string,lastName:string,email:string for your table

**Reply by IDrumsey** [reply]
That's doing it manually with --attributes.

**Reply by GreenFlag31** [reply]
then I'm afraid you have to do it manually :)
or create a small script that is going to analyze your file and create dynamically the sequelize-cli command and execute it, you to see :)

**Reply by IDrumsey** [reply]
Are you sure there is no other way to do it automatically? Otherwise I will leave this discussion open in case someone else knows.

**Reply by ennioVisco** [reply]
The idea of sequelize is to write the migrations manually, while the actual models can be auto-generated via https://github.com/sequelize/sequelize-auto. 
Perhaps you were trying to do the opposite, but if you think at it, it's much harder to start from the model and generate the migrations because it would be hard to tell in the code which is the old version of the model and which is the new one!
